lib/ddtrace/contrib/status_code_matcher.rb

```diff
@@ -0,0 +1,65 @@
+require 'set'
+require 'ddtrace/ext/http'
+
+module Datadog
+  module Contrib
+    # Contains methods helpful for tracing/annotating HTTP request libraries
+    class StatusCodeMatcher
+      REGEX_PARSER = /^\d{3}(?:-\d{3})?(?:,\d{3}(?:-\d{3})?)*$/
+
+      def initialize(range)
+        @error_response_range = range
+      end
```

Comment:
if we call `set-range` here we can ensure that we define+cache all the instance variables at startup time, which is preferable to execution time imo
```suggestion
      set_range
      end
```